### PR TITLE
Add Jetpack bi-yearly products to `calypso-products` package.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
This PR adds Jetpack Security T1 Bi-Yearly and VaultPress Backup T1 Bi-Yearly products/plans to the `calypso-products` package.

Asana Task: 1201210512993648-as-1204659654942256/f

## Proposed Changes

* Added `PRODUCT_JETPACK_BACKUP_T1_BI_YEARLY` to `products-list.ts`.
* Added `PLAN_JETPACK_SECURITY_T1_BI_YEARLY` to `plans-list.ts`.
* Added above 2 constants ⬆️ to `jetpack.ts` file.
* Updated `calypso-products' package unit tests, so that all passing.

## Testing Instructions

- First, sandbox `public-api.wordpress.com`.
- In your sandbox, open file: `wp-content/admin-plugins/wpcom-billing/default-shopping-cart-json-serializer.php`.
- Comment out lines 1050 - 1052 and save file, so that we can view Jetpack **Bi-Yearly** products at checkout.  It should looks like this:
```
// if ( 'jetpack' === $product->product_type && $product->renews_biyearly() ) {
// 	return null;
// }
```
- Go to https://wordpress.com/checkout/jetpack/jetpack_backup_t1_bi_yearly and https://wordpress.com/checkout/jetpack/jetpack_security_t1_bi_yearly
- Notice that the billing term variation dropdown is **Not** showing and the "**Included with your purchase**" features are **Not** showing.  This PR fixes this..  (See "Before" screenshot below.)
- Now checkout this branch/PR and spin up Calypso (`yarn start`).
- Now reload the checkout pages listed above and verify the billing term variation dropdown **_is_** showing and the "**Included with your purchase**" features **are now showing**. (See "After" screenshot below.)
- Run: `yarn test-packages calypso-products` and verify all tests are passing.

### Screenshots

Screenshot example for Jetpack Security (10GB) Bi-yearly (slug: `jetpack_security_t1_bi_yearly`)

BEFORE | AFTER
--- | ---
Markup 2023-05-24 at 15 05 29 | Markup 2023-05-24 at 15 07 37

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-products/src/plans-list.tsx [hunk 1]
```diff
@@ -2859,6 +2865,27 @@ export const PLANS_LIST: Record< string, Plan | JetpackPlan | WPComPlan > = {
 		],
 	},
 
+	[ PLAN_JETPACK_SECURITY_T1_BI_YEARLY ]: {
+		...getPlanJetpackSecurityT1Details(),
+		...getBiAnnualTimeframe(),
+		getStoreSlug: () => PLAN_JETPACK_SECURITY_T1_BI_YEARLY,
+		getPathSlug: () => 'security-20gb-yearly',
```
Comment (IanRamosC) [inline packages/calypso-products/src/plans-list.tsx]: It seems like we have different paths for the yearly and monthly plans, do we need a new one for this new billing term?
I'm not sure where that is being used tho

File: packages/calypso-products/src/translations.tsx [hunk 2]
```diff
@@ -1239,6 +1244,6 @@ export const useJetpackStorageAmountTextByProductSlug = (): ( (
 				[ PLAN_JETPACK_SECURITY_T2_MONTHLY ]: ONE_TERABYTE,
 				[ PLAN_JETPACK_SECURITY_T2_YEARLY ]: ONE_TERABYTE,
 			}[ productSlug ] ),
-		[ TEN_GIGABYTES, HUNDRED_GIGABYTES, ONE_TERABYTE, THREE_TERABYTE, FIVE_TERABYTE ]
+		[ TEN_GIGABYTES, ONE_TERABYTE, HUNDRED_GIGABYTES, THREE_TERABYTE, FIVE_TERABYTE ]
```
Comment (IanRamosC) [inline packages/calypso-products/src/translations.tsx]: I'm not sure if this was intentionally changed, but the previous order looks more right (10gb, 100gb, 1tb...)

**Review by IanRamosC — APPROVED** [review]
The change looks good and works as expected. I left some questions in the comments.

File: packages/calypso-products/src/translations.tsx (hunk 2 above)
Comment (elliottprogrammer) [reply packages/calypso-products/src/translations.tsx]: Opps! Yeah that change was not intentional. I'll change it back the way it was. 👍  Thanks Ian. 🙂

File: packages/calypso-products/src/plans-list.tsx (hunk 1 above)
Comment (elliottprogrammer) [reply packages/calypso-products/src/plans-list.tsx]: Thanks for reviewing Ian!!   Yeah, it's strange, it looks like `getPathSlug()` is only being used in a couple of places, and only for wpcom products, so for Jetpack products this value is not used or associated with anything.  I'll change it to match format of all other Jetpack products. 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8010416

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8010416).

**Pull request merged**